lib/host/aws.js

```diff
@@ -49,33 +51,44 @@ Read AWS metadata and user-data to build a configuration for the worker.
 @param {String} [baseUrl] optional base url override (for tests).
 @return {Object} configuration values.
 */
-function* configure (baseUrl) {
+export async function configure(baseUrl) {
   baseUrl = baseUrl || BASE_URL;
   log('configure', { url: BASE_URL });
 
   // defaults per the metadata
-  var config = yield {
-    host: getText(baseUrl + '/meta-data/public-hostname'),
-    // Since this is aws configuration after all...
+  let metadata = await Promise.all([
+    // host
+    getText(baseUrl + '/meta-data/public-hostname'),
+    // workerId
+    getText(baseUrl + '/meta-data/instance-id'),
+    // workerType
+    getText(baseUrl + '/meta-data/ami-id'),
+    // workerGroup
+    getText(baseUrl + '/meta-data/placement/availability-zone'),
+    // workerNodeType
+    getText(baseUrl + '/meta-data/instance-type')
+  ]);
```

Comment:
There is no real need for parallel requests here... We only do this one startup...
so you have been off fine with `await getText(baseUrl + '/meta-data/public-hostname')` for all the config entries..

Sometimes I wish we had things like`.spread` and `.props`: https://github.com/petkaantonov/bluebird/blob/master/API.md#props---promise

Anyways, this is very good :) Don't change it!